qualtran/bloqs/swap_network.py

```diff
@@ -125,3 +151,12 @@ def build_composite_bloq(
                 )
 
         return {'selection': bb.join(selection), 'targets': targets}
+
+    def bloq_counts(
+        self, ssa: Optional['SympySymbolAllocator'] = None
+    ) -> Set[Tuple[Union[int, sympy.Expr], Bloq]]:
+        num_swaps = 0
+        for j in range(self.selection_bitsize):
+            for i in range(0, self.n_target_registers - 2**j, 2 ** (j + 1)):
+                num_swaps += 1
```

Comment:
can we not write this more analytically?

Reply:
Done.